### ISSUE TITLE
Add support to load yaml files

### ISSUE THREAD

**Issue by horahh** [pola-rs/polars#10185]
### Problem description

Would be useful to directly load yaml files similar to csv or parquet.

**Comment by alexander-beedie** [comment]
Errr... aren't these usually just small configuration files, or is there a use-case I haven't seen before? Genuinely curious - loading data out of yaml has literally never occurred to me; can you provide more details, and maybe an example use-case?

**Comment by horahh** [comment]
I tend to agree, I think probably wont be widespread, but essentially the use case I had in mind would be a table from a hierarchy within a yaml.
